File: src/com/boardgamegeek/provider/GamesRanksIdProvider.java
package com.boardgamegeek.provider;

import android.net.Uri;

import com.boardgamegeek.provider.BggContract.GameRanks;
import com.boardgamegeek.provider.BggDatabase.Tables;
import com.boardgamegeek.util.SelectionBuilder;

public class GamesRanksIdProvider extends BaseProvider {

	@Override
	protected SelectionBuilder buildSimpleSelection(Uri uri) {
		int rankId = GameRanks.getRankId(uri);
		return new SelectionBuilder().table(Tables.GAME_RANKS).whereEquals(GameRanks.GAME_RANK_ID, rankId);
	}

	@Override
	protected String getPath() {
		return "games/ranks/#";
	}

	@Override
	protected String getType(Uri uri) {
		return GameRanks.CONTENT_ITEM_TYPE;
	}
}

File: app/src/main/java/com/boardgamegeek/provider/GamesRanksIdProvider.kt
package com.boardgamegeek.provider

import android.net.Uri
import com.boardgamegeek.provider.BggContract.*
import com.boardgamegeek.provider.BggContract.Companion.PATH_GAMES
import com.boardgamegeek.provider.BggContract.Companion.PATH_RANKS
import com.boardgamegeek.provider.BggDatabase.Tables

class GamesRanksIdProvider : BaseProvider() {
    override fun getType(uri: Uri) = GameRanks.CONTENT_ITEM_TYPE

    override val path = "$PATH_GAMES/$PATH_RANKS/#"

    override fun buildSimpleSelection(uri: Uri): SelectionBuilder {
        val rankId = GameRanks.getRankId(uri)
        return SelectionBuilder()
            .table(Tables.GAME_RANKS)
            .whereEquals(GameRanks.Columns.GAME_RANK_ID, rankId)
    }
}
